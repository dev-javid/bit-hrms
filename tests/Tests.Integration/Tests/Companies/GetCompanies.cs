namespace Tests.Integration.Tests.Companies
{
    public class GetCompanies(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
    {
        private const string Route = "/api/companies";

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Should_Return_Data()
        {
            await LoginAsSuperAdminAsync();
            await FeedDataAsync("Tests/Companies/GetCompanies.sql");
            using HttpResponseMessage response = await Client.GetAsync($"{Route}?page=1&limit=5");

            response.StatusCode.Should().Be(HttpStatusCode.OK);
            var result = await response.Content.ReadAsStringAsync();
            Approvals.VerifyJson(result);
        }
    }
}
