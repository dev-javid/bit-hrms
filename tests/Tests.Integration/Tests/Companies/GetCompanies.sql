INSERT INTO public.companies (id, name, email, phone_number, financial_month, administrator_name, weekly_off_days, is_deleted, created_by, created_on)
VALUES 
(1,  'Test Company 1',  'test1@company.com', '1234567890', 1, 'Admin 1', ARRAY['Sunday'], false, 1, now()),
(2,  'Test Company 2',  'test2@company.com', '2345678901', 2, 'Admin 2', ARRAY['Sunday'], false, 1, now()),
(3,  'Test Company 3',  'test3@company.com', '3456789012', 3, 'Admin 3', ARRAY['Sunday'], false, 1, now()),
(4,  'Test Company 4',  'test4@company.com', '4567890123', 4, 'Admin 4', ARRAY['Sunday'], false, 1, now()),
(5,  'Test Company 5',  'test5@company.com', '5678901234', 5, 'Admin 5', ARRAY['Sunday'], false, 1, now()),
(6,  'Test Company 6',  'test6@company.com', '6789012345', 6, 'Admin 6', ARRAY['Sunday'], false, 1, now()),
(7,  'Test Company 7',  'test7@company.com', '7890123456', 7, 'Admin 7', ARRAY['Sunday'], false, 1, now()),
(8,  'Test Company 8',  'test8@company.com', '8901234567', 8, 'Admin 8', ARRAY['Sunday'], false, 1, now()),
(9,  'Test Company 9',  'test9@company.com', '9012345678', 9, 'Admin 9', ARRAY['Sunday'], false, 1, now()),
(10, 'Test Company 10', 'test9@company.com', '9012345678', 9, 'Admin 9', ARRAY['Sunday'], true,  1, now());
