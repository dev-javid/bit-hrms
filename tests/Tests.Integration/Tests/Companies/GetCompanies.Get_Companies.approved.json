{
    "page": 1,
    "limit": 20,
    "total": 10,
    "items": [
        {
            "companyId": 1,
            "name": "Test Company 1",
            "phoneNumber": "1234567890",
            "financialMonth": 1,
            "administratorName": "Admin 1",
            "email": "test1@company.com",
            "address": "Test Address 1"
        },
        {
            "companyId": 2,
            "name": "Test Company 2",
            "phoneNumber": "2345678901",
            "financialMonth": 2,
            "administratorName": "Admin 2",
            "email": "test2@company.com",
            "address": "Test Address 2"
        },
        {
            "companyId": 3,
            "name": "Test Company 3",
            "phoneNumber": "3456789012",
            "financialMonth": 3,
            "administratorName": "Admin 3",
            "email": "test3@company.com",
            "address": "Test Address 3"
        },
        {
            "companyId": 4,
            "name": "Test Company 4",
            "phoneNumber": "4567890123",
            "financialMonth": 4,
            "administratorName": "Admin 4",
            "email": "test4@company.com",
            "address": "Test Address 4"
        },
        {
            "companyId": 5,
            "name": "Test Company 5",
            "phoneNumber": "5678901234",
            "financialMonth": 5,
            "administratorName": "Admin 5",
            "email": "test5@company.com",
            "address": "Test Address 5"
        },
        {
            "companyId": 6,
            "name": "Test Company 6",
            "phoneNumber": "6789012345",
            "financialMonth": 6,
            "administratorName": "Admin 6",
            "email": "test6@company.com",
            "address": "Test Address 6"
        },
        {
            "companyId": 7,
            "name": "Test Company 7",
            "phoneNumber": "7890123456",
            "financialMonth": 7,
            "administratorName": "Admin 7",
            "email": "test7@company.com",
            "address": "Test Address 7"
        },
        {
            "companyId": 8,
            "name": "Test Company 8",
            "phoneNumber": "8901234567",
            "financialMonth": 8,
            "administratorName": "Admin 8",
            "email": "test8@company.com",
            "address": "Test Address 8"
        },
        {
            "companyId": 9,
            "name": "Test Company 9",
            "phoneNumber": "9012345678",
            "financialMonth": 9,
            "administratorName": "Admin 9",
            "email": "test9@company.com",
            "address": "Test Address 9"
        },
        {
            "companyId": 999,
            "name": "Test Company",
            "phoneNumber": "9876543210",
            "financialMonth": 3,
            "administratorName": "Super Admin",
            "email": "company1@example.com",
            "address": "Test Address"
        }
    ]
}